packages/renderer/src/lib/kube/pods/PodDetails.svelte

```diff
@@ -0,0 +1,101 @@
+<script lang="ts">
+import type { V1Pod } from '@kubernetes/client-node';
+import { StatusIcon, Tab } from '@podman-desktop/ui-svelte';
+import { onMount } from 'svelte';
+import { router } from 'tinro';
+import { stringify } from 'yaml';
+
+import { kubernetesCurrentContextPods } from '/@/stores/kubernetes-contexts-state';
+
+import Route from '../../../Route.svelte';
+import MonacoEditor from '../../editor/MonacoEditor.svelte';
+import PodIcon from '../../images/PodIcon.svelte';
+import DetailsPage from '../../ui/DetailsPage.svelte';
+import StateChange from '../../ui/StateChange.svelte';
+import { getTabUrl, isTabSelected } from '../../ui/Util';
+import KubeEditYaml from '../KubeEditYAML.svelte';
+import KubernetesTerminalBrowser from './KubernetesTerminalBrowser.svelte';
+import { PodUtils } from './pod-utils';
+import PodActions from './PodActions.svelte';
+import PodDetailsLogs from './PodDetailsLogs.svelte';
+import PodDetailsSummary from './PodDetailsSummary.svelte';
+import type { PodUI } from './PodUI';
+
+interface Props {
+  name: string;
+  namespace: string;
+}
+let { name, namespace }: Props = $props();
+
+let pod = $state<PodUI | undefined>();
```

Comment:
You don't need to repeat have `| undefined` because it is part of $state() definition 
```
declare function $state<T>(initial: T): T;
declare function $state<T>(): T | undefined;
```
and the return value is `<T | undefined>` anyway, so we can spare four `| undefined` tokens.
`pnpm typecheck:renderer` is fine with that.

Reply:
Thanks, good catch & fixed. I thought it looked odd at the time but couldn't place why. :)